drivers/display/display_eadogs164.c

```diff
@@ -0,0 +1,464 @@
+/*
+ * Copyright (c) 2020
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ea_eadogs164
+
+#include <device.h>
+#include <drivers/i2c.h>
+#include <kernel.h>
+#include <sys/__assert.h>
+#include <logging/log.h>
+#include <string.h>
+#include <drivers/display.h>
+
+#include "display_eadogs164.h"
+
+struct eadogs164_display_data eadogs0_display_driver;
```

Comment:
Typically we write drivers for display controllers not for a display, therefore the driver should be written for SSD1803A controller and not for a DOGS164e display only.